### PR TITLE
[DOCS] Long table cell formatting fix

### PR DESCRIPTION
**Pull request by ChrisChinchilla**
### Description

This PR is in reference to https://github.com/ethereum/solidity/issues/4806 and replaces hyphen with non-breaking hyphen to fix wraps on longer lines.

### Checklist
- [x] Code compiles correctly
- [x] All tests are passing
- [x] New tests have been created which fail without the change (if possible)
- [x] README / documentation was extended, if necessary
- [x] Changelog entry (if change is visible to the user)
- [x] Used meaningful commit messages

### CODE REVIEW

**Comment by ChrisChinchilla** [comment]
Someone just told me of another (maybe better) solution, so don't review yet.

**Comment by ChrisChinchilla** [comment]
Here's a new option using the "official" Rst way of forcing a linebreak, plus a styling fix. But still it involves a lot of small changes for maybe what is quite a small issue.

**Comment by ChrisChinchilla** [comment]
Search for 

> call contract at address a with input mem

To see the row in question with a fix applied.

**Comment by ChrisChinchilla** [comment]
Now ready for proper review.

File: docs/yul.rst [hunk 1]
```diff
@@ -39,7 +39,7 @@ and ``mod`` are available either natively or as functions and computes exponenti
 .. code::
 
     {
-        function power(base:u256, exponent:u256) -> result:u256
+        function power(base:u256, exponent:u256) ‑> result:u256
```
Comment (chriseth) [inline docs/yul.rst]: Let's not do it here - it will not work for the compiler.

**Review by chriseth — CHANGES_REQUESTED** [review]
Please only change it in the table.

File: docs/yul.rst (hunk 1 above)
Comment (ChrisChinchilla) [reply docs/yul.rst]: Ah yes, sorry!

**Review by erak — APPROVED** [review]

**Pull request merged**